pkg/controller/secretgenerator/secretgenerator_controller.go

```diff
@@ -0,0 +1,139 @@
+func AddToManager(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// add adds a new controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New(
+		"secretgenerator-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch only new secrets with the corresponding annotation
+	predicates := predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			object, _ := meta.Accessor(e.Object)
+			if _, found := object.GetAnnotations()[SECRET_NAME_ANNOTATION]; found {
+				return true
+			}
+			return false
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			return false
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return false
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return false
+		},
+	}
+
+	// Watch for changes to Kubernetes secrets
+	err = c.Watch(&source.Kind{Type: &v1.Secret{}}, &handler.EnqueueRequestsFromMapFunc{
+		ToRequests: handler.ToRequestsFunc(func(a handler.MapObject) []reconcile.Request {
+			namespacedName := types.NamespacedName{Name: a.Meta.GetName(), Namespace: a.Meta.GetNamespace()}
+			return []reconcile.Request{{NamespacedName: namespacedName}}
+		}),
+	}, predicates)
+	if err != nil {
+		return err
+	}
+
+	log.Infof("Controller added to watch secrets with known GVK.")
+	return nil
+}
+
+// Reconcile will generate new secret with random data for the annotated secret
+// based on the specified type and complexity. This will avoid possible race
+// conditions when a deployment mounts the secret before it is reconciled
+func (r *ReconcileSecretGenerator) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	foundSecret := &v1.Secret{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, foundSecret)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
+
+	// Generate the secret if it does not previously exist
+	generatedSecret := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      foundSecret.Name + "-generated",
+			Namespace: foundSecret.Namespace,
+			Labels:    foundSecret.Labels,
+			OwnerReferences: []metav1.OwnerReference{
+				*metav1.NewControllerRef(foundSecret, foundSecret.GroupVersionKind()),
+			},
+		},
+	}
+
+	generatedSecretKey := types.NamespacedName{
+		Name: generatedSecret.Name, Namespace: generatedSecret.Namespace}
+	err = r.client.Get(context.TODO(), generatedSecretKey, generatedSecret)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			// Generate secret random value
+			log.Infof("Generating a random value for %v secret in the %v namespace",
+				generatedSecret.Name, generatedSecret.Namespace)
+
+			secret, err := newSecret(foundSecret.GetAnnotations())
+			if err != nil {
+				return reconcile.Result{}, err
```

Comment:
Errors defined in secret.go are not being logged during reconcile. We can add a log.error before `return`:
```
log.Errorf("error creating secret: %v", err)
return reconcile.Result{}, err
``` 

Current operator logs:
<img width="1412" alt="Screen Shot 2022-09-26 at 3 24 03 PM" src="https://user-images.githubusercontent.com/17230536/192250708-f5cbbdeb-7840-4dec-98eb-23b789d67930.png">

After the log statement:
<img width="1189" alt="Screen Shot 2022-09-26 at 3 34 34 PM" src="https://user-images.githubusercontent.com/17230536/192250771-68dc1409-c0da-466a-af16-9a2e93896853.png">

Reply:
Oh thanks for that, added!